src/Analysis/Ast/Impl/Analyzer/Symbols/SymbolCollector.cs

```diff
@@ -104,14 +115,16 @@ internal sealed class SymbolCollector : PythonWalker {
             }
         }
 
-        private void AddFunction(FunctionDefinition fd, IPythonType declaringType) {
-            if (!(_eval.LookupNameInScopes(fd.Name, LookupOptions.Local) is PythonFunctionType existing)) {
-                existing = new PythonFunctionType(fd, declaringType, _eval.GetLocationOfName(fd));
+        private void AddFunction(FunctionDefinition fd, PythonType declaringType) {
+            if (!(_eval.LookupNameInScopes(fd.Name, LookupOptions.Local) is PythonFunctionType f)) {
+                f = new PythonFunctionType(fd, declaringType, _eval.GetLocationOfName(fd));
                 // The variable is transient (non-user declared) hence it does not have location.
                 // Function type is tracking locations for references and renaming.
-                _eval.DeclareVariable(fd.Name, existing, VariableSource.Declaration);
+                _eval.DeclareVariable(fd.Name, f, VariableSource.Declaration);
+                _typeMap[fd] = f;
+                declaringType?.AddMember(f.Name, f, overwrite: true);
```

Comment:
Better handle functions with members (i.e. classes and functions inside functions)